addons/l10n_ec/models/account_move.py

```diff
@@ -175,17 +175,15 @@ def _get_l10n_ec_documents_allowed(self, identification_code):
 
     def _get_l10n_latam_documents_domain(self):
         self.ensure_one()
-        if self.journal_id.company_id.account_fiscal_country_id != self.env.ref('base.ec') or not \
-                self.journal_id.l10n_latam_use_documents:
-            return super()._get_l10n_latam_documents_domain()
-        domain = [
-            ('country_id.code', '=', 'EC'),
-            ('internal_type', 'in', ['invoice', 'debit_note', 'credit_note', 'invoice_in'])
-        ]
-        internal_type = self.l10n_latam_document_type_id.internal_type
-        allowed_documents = self._get_l10n_ec_documents_allowed(self._get_l10n_ec_ats_identification_type())
-        if internal_type and allowed_documents:
-            domain.append(("id", "in", allowed_documents.filtered(lambda x: x.internal_type == internal_type).ids))
+        domain = super(AccountMove, self)._get_l10n_latam_documents_domain()
+        if self.country_code == 'EC' and self.journal_id.l10n_latam_use_documents:
+            if self.debit_origin_id: # show/hide the debit note document type
+                domain.extend([("internal_type", "=", 'debit_note')])
+            elif self.move_type in ('out_invoice','in_invoice'):
+                domain.extend([("internal_type", "=", 'invoice')])
+            allowed_documents = self._get_l10n_ec_documents_allowed(self._get_l10n_ec_ats_identification_type())
```

Comment:
@pepetreshere @cormaza The Documents mapping is done in a too complicated way.  09, 20 and 21 are not even possible.  It should only depend on ruc / dni / consumidor final o exterior.  (and in the case of ruc everything can be possible)  The domain is mainly there to make it easier for the user.